core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -301,6 +303,116 @@ class TransactionStateManager(brokerId: Int,
       throw new KafkaException(s"Transaction topic number of partitions has changed from $transactionTopicPartitionCount to $curTransactionTopicPartitionCount")
   }
 
+  def appendTransactionToLog(transactionalId: String,
+                             txnMetadata: TransactionMetadata,
+                             responseCallback: Errors => Unit) {
+
+    // generate the message for this transaction metadata
+    val keyBytes = TransactionLog.keyToBytes(transactionalId)
+    val valueBytes = TransactionLog.valueToBytes(txnMetadata)
+    val timestamp = time.milliseconds()
+
+    val records = MemoryRecords.withRecords(TransactionLog.EnforcedCompressionType, new SimpleRecord(timestamp, keyBytes, valueBytes))
```

Comment:
I initially thought `SimpleRecord` would be used primarily in testing, but it has turned out to be useful generally to avoid depending on a particular message format version.